backend/dataall/modules/maintenance/services/maintenance_service.py

```diff
@@ -0,0 +1,119 @@
+"""
+A service layer for maintenance activity
+Defines functions and business logic to be performed for maintenance window
+"""
+
+import logging
+import os
+
+from dataall.base.aws.event_bridge import EventBridge
+from dataall.base.aws.parameter_store import ParameterStoreManager
+from dataall.modules.maintenance.api.enums import MaintenanceStatus
+from dataall.modules.maintenance.db.maintenance_repository import MaintenanceRepository
+from dataall.core.stacks.aws.ecs import Ecs
+
+logger = logging.getLogger(__name__)
+
+
+class MaintenanceService:
+    # Update the RDS table with the mode and status to PENDING
+    # Disable all scheduled ECS tasks which are created by data.all
+    @staticmethod
+    def start_maintenance_window(engine, mode: str = None):
+        logger.info('Putting data.all into maintenance')
+        try:
+            with engine.scoped_session() as session:
+                maintenance_record = MaintenanceRepository(session).get_maintenance_record()
+                if (
+                    maintenance_record.status == MaintenanceStatus.PENDING.value
+                    or maintenance_record.status == MaintenanceStatus.ACTIVE.value
+                ):
+                    logger.error(
+                        'Maintenance window already in PENDING or ACTIVE state. Cannot start maintenance window. Stop the maintenance window and start again'
+                    )
+                    return False
+                MaintenanceRepository(session).save_maintenance_status_and_mode(
+                    maintenance_status=MaintenanceStatus.PENDING.value, maintenance_mode=mode
+                )
+            # Disable scheduled ECS tasks
+            # Get all the SSM Params related to the scheduled tasks
+            ecs_scheduled_rules = ParameterStoreManager.get_parameters_by_path(
+                region=os.getenv('AWS_REGION', 'eu-west-1'),
```

Comment:
Should we really have a default value of eu-west-1 here?  Seems like there should be no default and that we check region value and error if it is not a valid region.

Reply:
I am taking the default value from the other similar boto3 wrappers which are defined in dataall/base/aws/ folder. I am in agreement that we should through error if the AWS_REGION env var is nt set. Maybe we should do it for other boto wrappers. 

@dlpzx , what are you views ? Is there any particular reason to keep the eu-west-1